docs/content/guides/dagster/managing-ml.mdx

```diff
@@ -0,0 +1,266 @@
+    reg = linear_model.LinearRegression()
+    reg.fit(training_data)
+    new_model_accuracy = reg.score(test_data)
+    if new_model_accuracy > model_accuracy:
+        yield Output(reg, metadata={"model_accuracy": new_model_accuracy})
+```
+
+We can also set up a sensor that triggers if an asset fails to materialize. Alerts can be customized and sent through e-mail or natively through Slack. In this example, a Slack message is sent anytime the `ml_job` fails.
+
+```python file=/guides/dagster/managing_ml/managing_ml_code.py startafter=fail_slack_start   endbefore=fail_slack_end
+import os
+from dagster import make_slack_on_run_failure_sensor, define_asset_job
+
+ml_job = define_asset_job("ml_training_job", selection=[ml_model])
+
+slack_on_run_failure = make_slack_on_run_failure_sensor(
+    channel="#ml_monitor_channel",
+    slack_token=os.getenv("MY_SLACK_TOKEN"),
+    monitored_jobs=(["ml_job"]),
+)
+```
+
+## Enhancing the Dagster UI with metadata
+
+Understanding the performance of your ML model is critical to both the model development process and production. Metadata can significantly enhance the usability of the Dagster UI to show what’s going on in a specific asset. Using metadata in Dagster is flexible, can be used for tracking evaluation metrics, and viewing the training accuracy progress over training iterations as a graph.
+
+One of the easiest ways to utilize Dagster’s metadata is by using a dictionary to track different metrics that are relevant for an ML model.
+
+Another way is to store relevant data for a single training iteration as a graph that you can view directly from the Dagster UI. In this example, we’ll define a function that uses data produced by a machine learning model to plot an evaluation metric as the model goes through the training process and render that in the Dagster UI.
+
+Dagster’s [`MetadataValue` types](/\_apidocs/ops#event-metadata>) enables types such as tables, URLs, notebooks and Markdown, etc. In the following example, the Markdown metadata type is used to generate plots. Each plot will show a specific evaluation metric’s performance throughout each training iteration also known as an epoch during the training cycle.
+
+```python file=/guides/dagster/managing_ml/managing_ml_code.py startafter=ui_plot_start   endbefore=ui_plot_end
+from dagster import MetadataValue
+import seaborn
+import matplotlib.pyplot as plt
+import base64
+from io import BytesIO
+
+
+def make_plot(eval_metric):
+    plt.clf()
+    training_plot = seaborn.lineplot(eval_metric)
+    fig = training_plot.get_figure()
+    buffer = BytesIO()
+    fig.savefig(buffer)
+    image_data = base64.b64encode(buffer.getvalue())
+    return MetadataValue.md(f"![img](data:image/png;base64,{image_data.decode()})")
+```
+
+Using the graph and other evaluation data in a machine learning model asset. In this example, we’re using a dictionary called metadata to store the Markdown plots and the score value in Dagster.
```

Comment:
> Using the graph and other evaluation data in a machine learning model asset.

This guy is an incomplete sentence and is just sorta hanging out. Can we drop it or re-word it to be grammatically correct, if you'd like to keep it?

Reply:
dropped it